### ISSUE TITLE
Suggestion: Super Quick Info to offer "Copy Fully-Qualified Symbol Name"

### ISSUE THREAD

**Issue by IneedHelp** [wmjordan/Codist#260]
As it is right now, we have the options to "Copy Symbol Name" and "Copy Qualified Symbol Name". 

Now, assuming the following property: `public static VM_MainWindow Instance { get; } = new();`

- if we hover over `Instance`, _Copy Symbol Name_ is pretty straight-forward, it copies `Instance `to clipboard (which one could argue that it's useful only if you want to perform copying by mouse movements, otherwise any symbol name can just be copied by double-click selection and Ctrl+C);
- if we hover over `Instance`, _Copy Qualified Symbol Name_ puts `VM_MainWindow.Instance` in the clipboard which might be useful in some cases, I'm not arguing that and it's a nice option to have;
- now what I am advocating for is a _Copy Fully-Qualified Symbol Name_ which would put in clipboard the full path to the symbol such as `MyRootNamespace.SRC.ViewModels.VM_MainWindow.Instance`. I find myself needing this symbol name copy operation more often than others, especially when I need to access static instances of a class in different parts of the application.

**Comment by IneedHelp** [comment]
Yes, an extra command to copy the fully qualified name of any symbol, but especially for members. And if I may offer another idea, maybe it could be useful to have quick access to these useful commands straight from the tooltip window located somewhere in the upper right corner (just their icons as buttons)

![image](https://github.com/wmjordan/Codist/assets/1366970/772273e4-d559-4a67-9bd3-2d8c23c9a294)

so that they can be quickly accessed without having to drill through the menu with extra clicks.

**Comment by wmjordan** [comment]
> maybe it could be useful to have quick access to these useful commands straight from the tooltip window located somewhere in the upper right corner (just their icons as buttons)

Great. I love that idea. Actually I have similar idea years ago. Maybe it is even possible to have a "toolbar" on Super Quick Info customizable too.

However, the Super Quick Info is window with a control somewhat like `StackPanel`, where absolute positioning is not supported. It is not so easy to put them into position, I am afraid.

**Comment by IneedHelp** [comment]
I too believe _Copy Symbol Name_ shouldn't make its way there; the _Go to Definition_ and the _Copy Qualified/Fully-Qualified Symbol Name_ commands would be great candidates for it until there's an ability to customize what goes there. As for the positioning, I was thinking upper-right corner alignment, so just plain relative alignment.

**Comment by wmjordan** [comment]
> the _Go to Definition_ and the _Copy Qualified/Fully-Qualified Symbol Name_ commands would be great candidates

_Go To Definition_? Click the symbol name nearby the big button on Super Quick Info. For instance, the `Instance` property name, at the right of the big button on your last screenshot.

<img src="https://user-images.githubusercontent.com/1366970/237944416-772273e4-d559-4a67-9bd3-2d8c23c9a294.png"/>

**Comment by IneedHelp** [comment]
Ah right, I need a bit more time to get used to all the available features. I just recently made the final decision to remove ReSharper enhanced tooltips which was previously making Super Quick Info non-interactive.

**Comment by wmjordan** [comment]
It is very easy to implement copying in the "namespaces.typeName.memberName" format.
What if reusing the "Copy Qualified Symbol Name" command--if you press Shift button, you get the fully-qualified format.
I can provide a tool tip to hint that behavior.

**Comment by IneedHelp** [comment]
> What if reusing the "Copy Qualified Symbol Name" command--if you press Shift button, you get the fully-qualified format. I can provide a tool tip to hint that behavior.
Thought about that as well and even about using middle-click as an alternative command on top of a base left-click command, but I also favor the plain point-and-click without having to remember any modifiers. I would suggest giving it it's own separate command to have a nice stack of _Copy Symbol Name_, _Copy Qualified Symbol Name_ and _Copy Fully-Qualified Symbol Name_ as the menu isn't really that crowded yet (in my opinion).

**Comment by wmjordan** [comment]
I don't want so many "Copy..." items in that menu, and "Copy Fully-Qualified Symbol Name" is scarcely used.

**Comment by IneedHelp** [comment]
You make the final call on if and how to implement this. This is just something I frequently use as part of a different extension

![image](https://github.com/wmjordan/Codist/assets/1366970/6c4c6cd3-371c-4510-a1ff-5533a856d873)

where I can get full names in a bar at the bottom of the editor. I just wanted Codist to feel more complete with this suggestion.

**Comment by wmjordan** [comment]
OK, I don't like pressing Shift very much.
But I don't want new entry just for this function. Otherwise I might stuff all "Copy" methods into a sub menu.

How about right clicking the entry?

**Comment by wmjordan** [comment]
Unfortunately `MenuItem` does not fire mouse events, except the `Click` event, which does not identify how this event is fired (left click, right click or keyboard enter).
Do you know any workaround to receive the right click event from a `MenuItem`?

**Comment by wmjordan** [comment]
Please test the new beta, which implemented this request and also adds some tool tips to the symbol analysis menu items.

**Comment by IneedHelp** [comment]
Got it, it's really good. Thank you for it!

**Comment by IneedHelp** [comment]
> > maybe it could be useful to have quick access to these useful commands straight from the tooltip window located somewhere in the upper right corner (just their icons as buttons)
> 
> Great. I love that idea. Actually I have similar idea years ago. Maybe it is even possible to have a "toolbar" on Super Quick Info customizable too.
> 

Is this something you're still considering or is it left aside for a while ?

**Comment by wmjordan** [comment]
I have difficulty in placing the toolbar to the right place.
And if it is possible, there should be another issue for it.
